packages/compiler-core/src/errors.ts

```diff
@@ -17,7 +17,7 @@ export function defaultOnWarn(msg: CompilerError) {
   __DEV__ && console.warn(`[Vue warn] ${msg.message}`)
 }
 
-type InferCompilerError<T> = T extends ErrorCodes
+export type InferCompilerError<T> = T extends ErrorCodes
```

Comment:
It cannot be reused, since it depends on `ErrorCodes`, which only needed by compiler-core.